modules/javafx.graphics/src/main/java/javafx/css/SimpleSelector.java

```diff
@@ -70,23 +70,7 @@ public String getName() {
     }
 
     /**
-     * Gets an immutable list of style-classes of the {@code Selector}.
-     * @return an immutable list of style-classes of the {@code Selector}
-     */
-    public List<String> getStyleClasses() {
```

Comment:
This is a public method, so removing it would be an incompatible API change. Given the decision to live with this public class, I think you should restore this method.